syntax = "proto3";

package sampler;

// The sampler service definition.
service Sampler {
  // Sends sub-graph from sampler to trainer
  rpc SendSubGraph (SamplerRequest) returns (SamplerReply) {}
}

// The sampler request message
message SamplerRequest {
  bytes vertices_id = 1;
  bytes layer = 2;
  bytes csr_data = 3;
  bytes csr_indices = 4;
  bytes csr_indptr = 5;
  int32 csr_shape_0 = 6;
  int32 csr_shape_1 = 7;
}

// The response message
message SamplerReply {
  string response = 1;
}